kano_i18n/init.py

```diff
@@ -9,8 +9,50 @@
 import gettext
 import __builtin__
 
+REGISTERED_DOMAINS = set()
+CURRENT_TRANSLATION = None
+
+
 def install(app, locale_dir=None):
+    global CURRENT_TRANSLATION
+
     # N_() defined globally for deferred translation
     __builtin__.__dict__['N_'] = lambda msg: msg
 
-    gettext.install(app, localedir=locale_dir, unicode=1)
+    trans = gettext.translation(app, localedir=locale_dir, fallback=True, codeset=None)
+    trans.install(True, names=None)
+
+    CURRENT_TRANSLATION = trans
+
+
+def register_domain(domain, locale_dir=None):
+    """
+    Register a domain as a fallback to the currently installed translation.
+    Used by library modules to register their translations on top of the current
+    app translation.
+    """
+    global REGISTERED_DOMAINS
+    global CURRENT_TRANSLATION
+
+    if not hasattr(__builtin__, '_') or not CURRENT_TRANSLATION:
+        # TODO
+        raise Exception('Localization has not been setup')
+
+    if domain in REGISTERED_DOMAINS:
+        # Already setup so we bail early here
+        return
+
+    # Create translation
+    trans = gettext.translation(
+        domain,
+        localedir=locale_dir,
+        fallback=True  # don't throw an error if the no .mo files found for domain, just return `NullTranslations`
+    )
+
+    CURRENT_TRANSLATION.add_fallback(trans)
```

Comment:
Yes you can. The relevant snippet it is:

```python
    def add_fallback(self, fallback):
        if self._fallback:
            self._fallback.add_fallback(fallback)
        else:
            self._fallback = fallback
```
https://hg.python.org/cpython/file/2.7/Lib/gettext.py#l187